### PR TITLE
- Allow control of throttlepolicy per session.

### PR DESCRIPTION
**Pull request by baldersheim**
- Move tests to same namespace as code to avoid polluting the public interface.
- Expose the throttlepolicy from SourceSession to avoid double bookeeping.

@bratseth and @jonmv PR
@vekterli FYI

### CODE REVIEW

File: documentapi/src/main/java/com/yahoo/documentapi/Parameters.java [hunk 1]
```diff
@@ -9,5 +13,12 @@
  * @author bratseth
  */
 public class Parameters {
-    // empty
+    ThrottlePolicy throttlePolicyOverride;
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/Parameters.java:16]: I suggest removing "override" from the names. It doesn't add anything.

**Review by jonmv — APPROVED** [review]
Nice!

**Review by bratseth — APPROVED** [review]
👍

**Comment by bratseth** [comment]
00:03:10 [INFO] --- abi-check-plugin:7-SNAPSHOT:abicheck (default) @ documentapi ---
00:03:10 [ERROR] Class com.yahoo.documentapi.Parameters: Extra method public void setThrottlePolicyOverride(com.yahoo.messagebus.ThrottlePolicy)
00:03:10 [ERROR] Class com.yahoo.documentapi.Parameters: Extra method public java.util.Optional getThrottlePolicyOverride()

File: documentapi/src/main/java/com/yahoo/documentapi/Parameters.java (hunk 1 above)
Comment (baldersheim) [reply documentapi/src/main/java/com/yahoo/documentapi/Parameters.java]: Did so, and that revealed use in subclass VisitorParameters.
I am not sure about binary compatibility. Does moving an interface to parent brake binary compatibility ?

Comment (bratseth) [reply documentapi/src/main/java/com/yahoo/documentapi/Parameters.java]: No, that is a compatible change.

**Pull request merged**